From 63b5a8b7e1376193beb873a77e2efd20619d0860 Mon Sep 17 00:00:00 2001
From: Wyon Bi <bivvy.bi@rock-chips.com>
Date: Mon, 23 Sep 2019 11:12:43 +0800
Subject: [PATCH] drm/dsi: Fix compilation when CONFIG_DRM_MIPI_DSI=n

drivers/gpu/drm/panel/panel-simple.o: In function `panel_simple_xfer_cmd_seq':
drivers/gpu/drm/panel/panel-simple.c:349: undefined reference to `mipi_dsi_device_transfer'
drivers/gpu/drm/panel/panel-simple.c:349:(.text+0xacc): relocation
truncated to fit: R_AARCH64_CALL26 against undefined symbol `mipi_dsi_device_transfer'

Fixes: 0f36136db14b ("drm/dsi: Export mipi_dsi_device_transfer()")
Change-Id: I34c6e4f732aa9c3fba5ef6d9ded012c53c8d107b
Signed-off-by: Wyon Bi <bivvy.bi@rock-chips.com>
---
 include/drm/drm_mipi_dsi.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/drm/drm_mipi_dsi.h b/include/drm/drm_mipi_dsi.h
index 4880f6c83e7d..5448b67a8a9c 100644
--- a/include/drm/drm_mipi_dsi.h
+++ b/include/drm/drm_mipi_dsi.h
@@ -247,8 +247,16 @@ enum mipi_dsi_dcs_tear_mode {
 #define MIPI_DSI_DCS_POWER_MODE_PARTIAL (1 << 5)
 #define MIPI_DSI_DCS_POWER_MODE_IDLE    (1 << 6)
 
+#ifdef CONFIG_DRM_MIPI_DSI
 ssize_t mipi_dsi_device_transfer(struct mipi_dsi_device *dsi,
 				 struct mipi_dsi_msg *msg);
+#else
+static inline ssize_t mipi_dsi_device_transfer(struct mipi_dsi_device *dsi,
+					       struct mipi_dsi_msg *msg)
+{
+	return -EINVAL;
+}
+#endif
 ssize_t mipi_dsi_dcs_write_buffer(struct mipi_dsi_device *dsi,
 				  const void *data, size_t len);
 ssize_t mipi_dsi_dcs_write(struct mipi_dsi_device *dsi, u8 cmd,
-- 
2.35.3

